src/commoncode/fileutils.py

```diff
@@ -83,7 +83,7 @@ def logger_debug(*args):
     logger.setLevel(logging.DEBUG)
 
     def logger_debug(*args):
-        return logger.debug(' '.join(isinstance(a, basestring) and a or repr(a) for a in args))
+        return logger.debug(' '.join(isinstance(a, str) and a or repr(a) for a in args))
```

Comment:
Why do not you use compat there too?

Reply:
This can also be used.If you want i can make changes to compat but i have already close that related PR.

Reply:
As both can be used because it passed on both version of python 2 and 3.

Reply:
We want to be consistent and always refer and use `compat` when we tests for strings. So please use compat here.

Reply:
Ok, i am doing.